testing/sdk_networks.py

```diff
@@ -15,6 +16,47 @@
 ENABLE_VIRTUAL_NETWORKS_OPTIONS = {"service": {"virtual_network_enabled": True}}
 
 
+def get_endpoint_names(package_name, service_name) -> list:
```

Comment:
Great to see refactoring of the endpoint handling @nickbp. I currently have a merge ready PR adding retries to the specific endpoint query which might be redundant with this work. #2629 Maybe you could reuse parts of this: https://github.com/mesosphere/dcos-commons/pull/2629/files#diff-462f3b69538152069336f48317e72544